File: MD.CMS.BusinessLogic.Core/DataAccess/Controllers/V2/Options/Interfaces/ISortableRequestOptions.cs
using MD.Tools.Helpers.Core.TypeAttributes;
using System;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Controllers.V2.Options.Interfaces
{
    public interface ISortableRequestOptions<T>
        where T : Enum
    {
        T SortField { get; }
        SortDirection SortDirection { get; }
    }

    public enum SortDirection
    {
        [StringValue("ASC")]
        Ascending,
        [StringValue("DESC")]
        Descending
    }
}

File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/controllers/options/v2/iSortableRequestOptions.ts
/// <reference path="./enums/contentEnum.ts" />

namespace mdBusinessLogic.dataAccess.controllers.options.v2 {
    export interface iSortableRequestOptions<T> {
        SortField?: T;
        SortDirection?: sortDirection;
    }

    export enum sortDirection {
        Ascending,
        Descending
    }
}
